backend/src/repositories/user.rs

```diff
@@ -8,32 +9,54 @@ use crate::{
 
 #[derive(Clone)]
 pub struct UserRepository {
-  config: Config,
+  gamma_client: GammaClient,
 }
 
 impl UserRepository {
   pub fn new(config: Config) -> Self {
-    Self { config }
+    let gamma_client = GammaClient::new(&config.gamma_config);
+    Self { gamma_client }
   }
 
-  pub async fn get(&self, id: String) -> HubbitResult<GammaUser> {
-    let client = Client::new();
-    let res = client
-      .get(&format!(
-        "{}/api/users/{}",
-        self.config.gamma_internal_url, id
-      ))
-      .header(
-        AUTHORIZATION,
-        format!("pre-shared {}", self.config.gamma_api_key),
-      )
-      .send()
-      .await?;
-    if res.status() == 404 {
-      return Err(HubbitError::NotFound);
+  pub async fn get(&self, id: &Uuid) -> HubbitResult<GammaUser> {
+    let user = self.gamma_client.get_user(id).await.map_err(|err| {
+      if matches!(err, GammaError::NotFoundResponse { .. }) {
+        return HubbitError::NotFound;
+      }
+
+      HubbitError::GammaError(err)
+    })?;
+
+    let groups = self.gamma_client.get_groups_for_user(&user.id).await?;
+
+    Ok(GammaUser::from_user_and_groups(user, groups))
+  }
+
+  pub async fn get_by_cid(&self, cid: &str) -> HubbitResult<GammaUser> {
+    log::warn!(
```

Comment:
This is only used as a fallback and I'm not sure if it ever occurs in a real-world use-case (unless someone manually removes it from the cache).